Doc/tools/version_next.py

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env python3
+"""
+Replace `.. versionchanged:: next` lines in docs files by the given version.
+
+Applies to all the VersionChange directives. For deprecated-removed, only
+handle the first argument (deprecation version, not the removal version).
+
```

Comment:
I am wondering why an external script is needed, instead of the sphinx extension doing the work!

Reply:
The script should be run at release time, replacing `next` with the just-released version directly in the ReST sources. Any `next` added after that will be for the *following* version.

The Sphinx extension replaces `next` in the output (HTML, etc.).

Reply:
``check-warnings.py`` is named with a hypen to help signal it is a script rather than a module, perhaps we could do the same here?

Reply:
That would complicate the test :)